streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -226,11 +226,23 @@ public class KafkaStreams implements AutoCloseable {
      *
      * </pre>
      * Note the following:
-     * - RUNNING state will transit to REBALANCING if any of its threads is in PARTITION_REVOKED or PARTITIONS_ASSIGNED state
-     * - REBALANCING state will transit to RUNNING if all of its threads are in RUNNING state
-     * - Any state except NOT_RUNNING, PENDING_ERROR or ERROR can go to PENDING_SHUTDOWN (whenever close is called)
-     * - Of special importance: If the global stream thread dies, or all stream threads die (or both) then
-     *   the instance will be in the ERROR state. The user will not need to close it.
+     * <ul>
+     *   <li>
+     *     RUNNING state will transit to REBALANCING if any of its threads is in PARTITION_REVOKED or PARTITIONS_ASSIGNED state
+     *   </li>
+     *   <li>
+     *     REBALANCING state will transit to RUNNING if all of its threads are in RUNNING state
```

Comment:
nit: Should we align this documentation with `maybeSetRunning`?

https://github.com/apache/kafka/blob/b480135b4f5342ae0c4b4df214efc02a402c610e/streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java#L650